assets/js/googlesitekit/datastore/site/urls.js

```diff
@@ -104,6 +104,70 @@ export const selectors = {
 		} );
 	} ),
 
+	/**
+	 * Gets the Site Kit documentation URL.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} slug  The slug of the documentation page.
+	 * @return {string} The Site Kit support URL.
+	 */
+	getDocumentationLinkURL: createRegistrySelector(
+		( select ) => ( state, slug ) => {
+			const proxySupportLink = select(
+				CORE_SITE
+			).getProxySupportLinkURL();
+
+			if ( ! slug ) {
+				return proxySupportLink;
+			}
+
+			return `${ proxySupportLink }?doc=${ encodeURIComponent( slug ) }`;
+		}
+	),
+
+	/**
+	 * Gets the relevant troubleshooting URL.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {Object} error The error object.
+	 * @return {string} The Site Kit support URL.
+	 */
+	getErrorTroubleshootingLinkURL: createRegistrySelector(
+		( select ) => ( state, error ) => {
+			function isNumber( n ) {
+				return ! isNaN( parseFloat( n ) ) && ! isNaN( n - 0 );
+			}
+
+			const proxySupportLink = select(
+				CORE_SITE
+			).getProxySupportLinkURL();
+
+			if ( ! error ) {
+				return proxySupportLink;
+			}
+
+			if ( error.code && ! isNumber( error?.code ) ) {
```

Comment:
You don't need to check for the optional chaining condition here since it will already be `true` in the previous condition `error.code`; otherwise, it won't execute the second condition. Preferably you could simplify it with:
```suggestion
			if ( ! isNumber( error?.code ) ) {
```

Reply:
The first condition in the if-statements (`error.code`) prevents the second condition from running as you said, which prevents the condition from returning `undefined` if it isn't set. The `isNumber` function from lodash just returns `undefined` if no argument is provided.

Reply:
Suppose you want to ensure that the `code` property is available with a value defined to pass it to the `isNumber` function; you can check like this without the optional chaining `if ( error.code && ! isNumber( error.code ) )` because it will execute the second condition if the first is `true`.